data_model/src/domain.rs

```diff
@@ -232,6 +236,15 @@ impl Domain {
         self.asset_definitions.get(asset_definition_id)
     }
 
+    /// Return a reference to the asset definition corresponding to the asset definition id
+    #[inline]
+    pub fn asset_total_quantity(
+        &self,
+        asset_definition_id: &<AssetDefinition as Identifiable>::Id,
+    ) -> Option<&NumericValue> {
```

Comment:
With the change to accept `AssetValueType::Store` for total amount, this one would also become non-optional and more ergonomic IMO.

Reply:
@ilchu, you mean that we can `.expect` inside function body and not outside?